### PR TITLE
10529 adv search list operators

### PR DESCRIPTION
**Pull request by whatisgalen**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

If you have N concept values you want to filter for in advanced search, for example "show me resources that have any of these values for Node 'thing type'" you'll have to apply N "or" facets using the "equals" operator and each concept value. 

Instead of applying N facets, this PR proposes using a single facet to filter for N concept values using select2 multiselect in conjunction with these operators:
- Has any value in list (`in_list_any`)
- Has all values in list (`in_list_all`)
- Has no values in list (`in_list_not`)

These proposed functionalities greatly expand what the advanced search can do in terms of more complex or wide-ranging queries.

### Issues Solved

#10529 

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: Los Angeles OHR
*   Found by: @whatisgalen 
*   Tested by: @chiatt 
*   Designed by: @whatisgalen

### CODE REVIEW

File: arches/app/datatypes/concept_types.py [hunk 1]
```diff
@@ -139,6 +145,27 @@ def append_search_filters(self, value, node, query, request):
         except KeyError as e:
             pass
 
+    def append_in_list_search_filters(self, value, node, query, request, match_any=True):
+        try:
+            # Extract the list of values from the filter
+            values_list = value.get("val", [])
+
+            if values_list:
+                field_name = "tiles.data.%s" % (str(node.pk))
```
[the remark this answers is not in the document]
Comment (whatisgalen) [reply arches/app/datatypes/concept_types.py:154]: requested change in latest

File: arches/app/datatypes/concept_types.py [hunk 2]
```diff
@@ -139,6 +145,27 @@ def append_search_filters(self, value, node, query, request):
         except KeyError as e:
             pass
 
+    def append_in_list_search_filters(self, value, node, query, request, match_any=True):
```
[the remark this answers is not in the document]
Comment (whatisgalen) [reply arches/app/datatypes/concept_types.py]: requested change in latest

File: arches/app/datatypes/concept_types.py [hunk 3]
```diff
@@ -139,6 +145,27 @@ def append_search_filters(self, value, node, query, request):
...
+        try:
+            # Extract the list of values from the filter
+            values_list = value.get("val", [])
+
+            if values_list:
+                field_name = "tiles.data.%s" % (str(node.pk))
+                for val in values_list:
+                    match_q = Match(field=field_name, type="phrase", query=val)
+
+                    if match_any:
```
[the remark this answers is not in the document]
Comment (whatisgalen) [reply arches/app/datatypes/concept_types.py]: requested change in latest

**Review by whatisgalen — COMMENTED** [review]
all requested changes in latest

**Review by chiatt — CHANGES_REQUESTED** [review]

**Review by chiatt — APPROVED** [review]

**Pull request merged**